### ISSUE TITLE
Infinite Scroll example crashes on Firefox

### ISSUE THREAD

**Issue by process0** [KevinVandy/mantine-react-table#17]
### mantine-react-table version

v0.9.3

### react & react-dom versions

Whatever is in the example

### Describe the bug and the steps to reproduce it

Scrolling in Firefox will result in crashing the component.  This occurs in the example online, but this output is from the example running locally.

```
10:17:10.180 Uncaught Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
    React 3
    onChange index.tsx:31
    notify index.ts:353
    _measureElement index.ts:588
    measureElement index.ts:597
    ref MRT_TableBodyRow.tsx:78
    React 14
    workLoop scheduler.development.js:266
    flushWork scheduler.development.js:239
    performWorkUntilDeadline scheduler.development.js:533
    js scheduler.development.js:571
    js scheduler.development.js:633
    __require chunk-DFKQJ226.js:8
    js index.js:6
    __require chunk-DFKQJ226.js:8
    React 2
    __require chunk-DFKQJ226.js:8
    js React
    __require chunk-DFKQJ226.js:8
    <anonymous> chunk-I56DSPBQ.js:6058
react-dom.development.js:27292:10
```

### Minimal, Reproducible Example - (Optional, but Recommended)

https://www.mantine-react-table.com/docs/examples/infinite-scrolling

### Do you intend to try to help solve this bug with your own PR?

No, because I do not know how

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by KevinVandy** [comment]
Apparently this only happens on Firefox on Windows/Linux when scrolling happens very fast.

**Comment by KevinVandy** [comment]
@process0  This may be fixed in v1.0.0-beta.7.
